addons/escoria-core/game/core-scripts/esc_item.gd

```diff
@@ -642,51 +673,91 @@ func set_angle(deg: int, wait: float = 0.0):
 # - deg: The angle degree to set
 # - float Wait this amount of seconds until continuing with turning around
 func turn_to(object: Node, wait: float = 0.0):
-	_movable.turn_to(object, wait)
+	if is_movable:
+		_movable.turn_to(object, wait)
+	else:
+		escoria.logger.warn(
+			self,
+			"Node %s cannot use \"turn_to\". Its \"is_movable\" parameter is false." %self
+		)
 
 
+# Check everything is in place to play talk animations
+func check_talk_possible():
+	if is_movable and (_movable.last_dir < 0 \
+			or _movable.last_dir >= animations.speaks.size()):
+		escoria.logger.warn(
+			self,
+			"Node %s cannot talk. Its \"last_dir\" parameter is invalid : %s." \
+			%[self, _movable.last_dir]
+		)
+		return false
+	if animations == null:
+		escoria.logger.warn(
+			self,
+			"Node %s cannot talk. Its \"animations\" parameter is empty : %s." \
+			%self
+		)
+		return false
+	if animations.speaks.size() == 0:
+		escoria.logger.warn(
+			self,
+			"Node %s cannot talk. Its \"animations.speaks\" array is empty : %s." \
+			%self
+		)
+		return false
+	if not get_animation_player():
+		escoria.logger.warn(
+			self,
+			"Node %s cannot talk. Its animation player can't be found." \
+			%self
+		)
+		return false
+	return true
+	
+	
 # Play the talking animation
 func start_talking():
-	# Only start the speaking animation if we actually have them setup
-	if animations != null \
-			and animations.speaks.size() > 0 \
-			and get_animation_player() \
-			and _movable.last_dir >= 0 \
-			and _movable.last_dir < animations.speaks.size():
-		var animation_player = get_animation_player()
+	if not check_talk_possible():
+		return
+		
+	var animation_player = get_animation_player()
 
-		if animation_player.is_playing():
-			animation_player.stop()
+	if animation_player.is_playing():
+		animation_player.stop()
 
-		if animations.speaks[_movable.last_dir].mirrored \
-				and not _movable.is_mirrored:
-			_sprite_node.scale.x *= -1
+	if is_movable and animations.speaks[_movable.last_dir].mirrored \
```

Comment:
This test of `is_movable` value is included in `check_talk_possible()`. Is it necessary again here?

Reply:
This one is necessary as you only want to check animations...mirrored when is_movable is true otherwise this check will fail.